diff --git a/xbmc/utils/EGLUtils.cpp b/xbmc/utils/EGLUtils.cpp
index d843f01d7f9..e485c0201ad 100644
--- a/xbmc/utils/EGLUtils.cpp
+++ b/xbmc/utils/EGLUtils.cpp
@@ -34,6 +34,9 @@ namespace
 #ifndef EGL_CONTEXT_PRIORITY_MEDIUM_IMG
 #define EGL_CONTEXT_PRIORITY_MEDIUM_IMG 0x3102
 #endif
+#ifndef TARGET_STB
+#define TARGET_STB 1
+#endif
 
 #define X(VAL) std::make_pair(VAL, #VAL)
 std::map<EGLint, const char*> eglAttributes =
