diff --git a/xbmc/cores/HiPlayer/HiPlayerVideo.cpp b/xbmc/cores/HiPlayer/HiPlayerVideo.cpp
index 812b37d1a1f..591b8755058 100644
--- a/xbmc/cores/HiPlayer/HiPlayerVideo.cpp
+++ b/xbmc/cores/HiPlayer/HiPlayerVideo.cpp
@@ -34,6 +34,7 @@
 #include <linux/dvb/video.h>
 
 #define MAX_PAYLOAD 4096
+#define SAFE_DELETE(p) if ((p) != NULL) { delete (p); (p) = NULL; }
 
 CHiPlayerVideo::CHiPlayerVideo(IHiPlayerCallback *callback, CProcessInfo &processInfo): m_callback(callback)
 	,m_processInfo(processInfo)
