diff --git a/lib/python/Plugins/Extensions/DLNAServer/plugin.py b/lib/python/Plugins/Extensions/DLNAServer/plugin.py
index 11a5cc3..6c80def 100644
--- a/lib/python/Plugins/Extensions/DLNAServer/plugin.py
+++ b/lib/python/Plugins/Extensions/DLNAServer/plugin.py
@@ -14,15 +14,11 @@ from Components.ActionMap import NumberActionMap, ActionMap
 from Components.config import config, ConfigSelection, getConfigListEntry, ConfigText, ConfigDirectory, ConfigYesNo, ConfigSubsection
 from Components.FileList import FileList
 
-from Tools.Directories import resolveFilename, SCOPE_PLUGINS
-
 config.plugins.dlnaserver = ConfigSubsection()
 config.plugins.dlnaserver.autostart = ConfigYesNo(default = False)
 
-runcherBin = resolveFilename(SCOPE_PLUGINS, "Extensions/DLNAServer/dlnaserver")
-
 def isRunning():
-	ps_str = os.popen('ps -ef | grep minidlna | grep -v grep').read()
+	ps_str = os.popen('ps | grep minidlna | grep -v grep').read()
 	if ps_str.strip() != '':
 		return True
 	return False
@@ -165,11 +161,11 @@ class DLNAServer(ConfigListScreen, Screen):
 			self.session.openWithCallback(self.cbChangeDirectory, SelectDirectoryWindow, currentItem.value)
 
 	def keyGreen(self):
-		args = '-e'
+		args = 'stop'
 		if self["key_green"].getText().strip() == 'Start':
-			args = '-s'
+			args = 'start'
 			self.saveConfigFile()
-		rc = os.popen('%s %s'%(runcherBin, args)).read()
+		rc = os.popen('/etc/init.d/minidlna %s'%(args)).read()
 		self["information"].setText(rc)
 		self.updateGreenTimer.start(1000)
 
@@ -364,12 +360,12 @@ def main(session, **kwargs):
 def autostart(reason, **kwargs):
 	if reason == 0:
 		if isRunning():
-			args = " -e"
+			args = "stop"
 			is_running = True
 		else:
-			args = " -s"
+			args = "start"
 			is_running = False
-		cmd = runcherBin + args
+		cmd = "/etc/init.d/minidlna " + args
 
 		if config.plugins.dlnaserver.autostart.value:
 			if is_running:
