--- a/lib/python/Plugins/Extensions/DLNAServer/plugin.py
+++ b/lib/python/Plugins/Extensions/DLNAServer/plugin.py
@@ -1,5 +1,5 @@
 from Plugins.Plugin import PluginDescriptor
-
+import socket
 import os
 from enigma import eTimer
 
@@ -18,7 +18,7 @@
 config.plugins.dlnaserver.autostart = ConfigYesNo(default = False)
 
 def isRunning():
-	ps_str = os.popen('ps | grep minidlna | grep -v grep').read()
+	ps_str = os.popen('ps | grep minidlnad | grep -v grep').read()
 	if ps_str.strip() != '':
 		return True
 	return False
@@ -38,7 +38,7 @@
 		"""
 	def __init__(self, session, currentDir):
 		Screen.__init__(self, session)
-		inhibitDirs = ["/bin", "/boot", "/dev", "/etc", "/lib", "/proc", "/sbin", "/sys", "/usr", "/var"]
+		inhibitDirs = ["/bin", "/boot", "/dev", "/etc", "/lib", "/proc", "/sbin", "/sys", "/usr"]
 		self["filelist"] = FileList(currentDir, showDirectories = True, showFiles = False, inhibitMounts=[], inhibitDirs=inhibitDirs)
 		self["actions"]  = ActionMap(["WizardActions", "DirectionActions", "ColorActions", "EPGSelectActions"], {
 			"back"  : self.cancel,
@@ -165,13 +165,13 @@
 		if self["key_green"].getText().strip() == 'Start':
 			args = 'start'
 			self.saveConfigFile()
-		rc = os.popen('/etc/init.d/minidlna %s'%(args)).read()
+		rc = os.popen('/etc/init.d/readymedia.sh %s'%(args)).read()
 		self["information"].setText(rc)
 		self.updateGreenTimer.start(1000)
 
 	def keyYellow(self):
 		self.saveConfigFile()
-		self["information"].setText('finished saving!!')
+		self["information"].setText('Finished saving')
 
 	def keyBlue(self):
 		self.menuItemServerName.value = self.oldConfig.get('friendly_name')
@@ -334,10 +334,7 @@
 					self.oldConfig[key] = default
 			except: self.oldConfig[key] = default
 			
-		try:
-			model = os.popen('cat /proc/stb/info/vumodel').read().strip()
-		except: model = 'My'
-		setDefault('friendly_name', '%s DLNA Server'%(model.upper()))
+		setDefault('friendly_name', '%s DLNA Server' % (socket.gethostname()))
 		setDefault('media_dirV', '/media/dlna/Videos')
 		setDefault('media_dirA', '/media/dlna/Musics')
 		setDefault('media_dirP', '/media/dlna/Pictures')
@@ -365,7 +362,7 @@
 		else:
 			args = "start"
 			is_running = False
-		cmd = "/etc/init.d/minidlna " + args
+		cmd = "/etc/init.d/readymedia.sh " + args
 
 		if config.plugins.dlnaserver.autostart.value:
 			if is_running:
@@ -378,5 +375,5 @@
 				os.system(cmd)
 
 def Plugins(**kwargs):
- 	return [PluginDescriptor(name="DLNA Server", description="This is dlna server using minidlna.", where = PluginDescriptor.WHERE_PLUGINMENU, needsRestart = False, fnc=main),
+ 	return [PluginDescriptor(name="DLNA Server", description="DLNA server using ReadyMedia", where = PluginDescriptor.WHERE_PLUGINMENU, needsRestart = False, fnc=main),
 		PluginDescriptor(where = [PluginDescriptor.WHERE_AUTOSTART], fnc = autostart)]
