diff -Naur gst-plugins-bad.orig/gst-libs/gst/insertbin/Makefile.am gst-plugins-bad/gst-libs/gst/insertbin/Makefile.am
--- a/gst-libs/gst/insertbin/Makefile.am	2018-03-21 13:45:50.868076825 +0100
+++ b/gst-libs/gst/insertbin/Makefile.am	2018-03-21 13:46:41.800753342 +0100
@@ -45,7 +45,7 @@
 		--library=libgstinsertbin-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-insertbin-@GST_API_VERSION@ \
diff -Naur gst-plugins-bad.orig/gst-libs/gst/mpegts/Makefile.am gst-plugins-bad/gst-libs/gst/mpegts/Makefile.am
--- a/gst-libs/gst/mpegts/Makefile.am	2018-03-21 13:45:50.869076799 +0100
+++ b/gst-libs/gst/mpegts/Makefile.am	2018-03-21 13:46:41.800753342 +0100
@@ -82,7 +82,7 @@
 		--add-include-path=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-video-@GST_API_VERSION@` \
 		--library=libgstmpegts-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-video-@GST_API_VERSION@ \
 		--pkg-export gstreamer-mpegts-@GST_API_VERSION@ \
