diff -Naur gst-plugins-bad.orig/ext/hls/m3u8.c gst-plugins-bad/ext/hls/m3u8.c
--- gst-plugins-bad.orig/ext/hls/m3u8.c	2018-03-21 13:45:50.831077787 +0100
+++ gst-plugins-bad/ext/hls/m3u8.c	2018-03-21 13:58:05.979982489 +0100
@@ -358,17 +358,17 @@
 
     if (f1->sequence == f2->sequence && !g_str_equal (f1->uri, f2->uri)) {
       /* Same sequence, different URI. This is bad! */
-      GST_ERROR ("Media URIs inconsistent (sequence %" G_GINT64_FORMAT
+      GST_ERROR ("Media URIs inconsistent, continue anyways (sequence %" G_GINT64_FORMAT
           "): had '%s', got '%s'", f1->sequence, f2->uri, f1->uri);
-      return FALSE;
+      //return FALSE;
     } else if (f1->sequence < f2->sequence) {
       /* Not same sequence but by construction sequence must be higher in the
        * new one. All good in that case, if it isn't then this means that
        * sequence numbers are decreasing or files were inserted */
-      GST_ERROR ("Media sequences inconsistent: %" G_GINT64_FORMAT " < %"
+      GST_ERROR ("Media sequences inconsistent, continue anyways: %" G_GINT64_FORMAT " < %"
           G_GINT64_FORMAT ": URIs new '%s' old '%s'", f1->sequence,
           f2->sequence, f1->uri, f2->uri);
-      return FALSE;
+      //return FALSE;
     }
   }
 
