diff --git a/os_dep/os_intfs.c b/os_dep/os_intfs.c
index aad11c9..6b5012c 100644
--- a/os_dep/os_intfs.c
+++ b/os_dep/os_intfs.c
@@ -902,18 +902,19 @@ unsigned int rtw_classify8021d(struct sk_buff *skb)
 	return dscp >> 5;
 }
 
-#if (LINUX_VERSION_CODE>=KERNEL_VERSION(4,19,0))
-static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb,
-				struct net_device *sb_dev,
-				select_queue_fallback_t fallback
-#else
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0))
-			    , void *unused
-#endif
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0))
-			    , select_queue_fallback_t fallback
-#endif
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,19,0))
+	, struct net_device *sb_dev
+	#if (LINUX_VERSION_CODE < KERNEL_VERSION(5,2,0))
+	, select_queue_fallback_t fallback
+	#endif
+#else
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0)
+	, void *accel_priv
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+	, select_queue_fallback_t fallback
+	#endif
+	#endif
 #endif
 )
 {
