diff -ur nss-3.64/nss/lib/ssl/ssl3exthandle.c nss-3.64/nss/lib/ssl/ssl3exthandle.c
--- nss-3.64/nss/lib/ssl/ssl3exthandle.c	2023-08-27 12:08:39.648679904 +0200
+++ nss-3.64/nss/lib/ssl/ssl3exthandle.c	2023-08-27 12:09:01.737344800 +0200
@@ -201,7 +201,7 @@
  * Clients sends a filled in session ticket if one is available, and otherwise
  * sends an empty ticket.  Servers always send empty tickets.
  */
-PRInt32
+SECStatus
 ssl3_ClientSendSessionTicketXtn(const sslSocket *ss, TLSExtensionData *xtnData,
                                 sslBuffer *buf, PRBool *added)
 {
