diff -ur nss-3.64/nss/lib/ssl/sslsnce.c nss-3.64/nss/lib/ssl/sslsnce.c
--- nss-3.64/nss/lib/ssl/sslsnce.c	2023-08-27 12:58:03.573335828 +0200
+++ nss-3.64/nss/lib/ssl/sslsnce.c	2023-08-27 12:57:46.788586225 +0200
@@ -1820,7 +1820,7 @@
     return SECSuccess;
 }
 
-static PRBool
+static SECStatus
 ssl_GenerateSelfEncryptKeys(void *pwArg, PRUint8 *keyName,
                             PK11SymKey **aesKey, PK11SymKey **macKey);
 
