diff -ur nss-3.64/nss/cmd/ecperf/ecperf.c nss-3.64/nss/cmd/ecperf/ecperf.c
--- nss-3.64/nss/cmd/ecperf/ecperf.c	2021-04-15 18:17:44.000000000 +0200
+++ nss-3.64/nss/cmd/ecperf/ecperf.c	2023-08-27 13:03:00.223915074 +0200
@@ -43,6 +43,11 @@
 
 typedef SECItem SECKEYECParams;
 
+#pragma GCC diagnostic push
+#if __GNUC__ >= 12 
+  #pragma GCC diagnostic ignored "-Wdangling-pointer"
+#endif
+
 void
 PKCS11Thread(void *data)
 {
@@ -112,6 +115,8 @@
     return;
 }
 
+#pragma GCC diagnostic pop
+
 /* Time iter repetitions of operation op. */
 SECStatus
 M_TimeOperation(void (*threadFunc)(void *),
