From 676c1fac877924b76d4a45e8f9999300cd2b6464 Mon Sep 17 00:00:00 2001
From: kueken <ohnemailadresse@arcor.de>
Date: Fri, 17 Nov 2023 22:00:52 +0100
Subject: [PATCH] add missing Cflag

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 9f2e1397ed..11a61654df 100644
--- a/configure.ac
+++ b/configure.ac
@@ -261,6 +261,7 @@ CXXFLAGS="$CXXFLAGS -Wall -Wextra -Werror \
 	-Wno-error=stringop-truncation -Wno-error=shadow -Wno-error=cast-qual \
 	-Wno-error=aggregate-return -Wno-error=missing-field-initializers \
 	-Wno-error=packed -Wno-error=vla -Wno-error=clobbered -Wno-error=unused-parameter \
+	-Wno-error=use-after-free -Wno-error=dangling-pointer= \
 	$BASE_CFLAGS $ENIGMA2_CFLAGS $PTHREAD_CFLAGS $OPENMP_CFLAGS $ALSA_CFLAGS"
 
 ARFLAGS="cr"
-- 
2.34.1

