diff --git a/lib/dvb/pmtparse.cpp b/lib/dvb/pmtparse.cpp
index fc1e2a51b..8b54cd58b 100644
--- a/lib/dvb/pmtparse.cpp
+++ b/lib/dvb/pmtparse.cpp
@@ -494,11 +494,11 @@ DEFINE_REF(eDVBPMTParser::eStreamData);
 
 eDVBPMTParser::eStreamData::eStreamData(eDVBPMTParser::program &program)
 {
-	for (const auto i : program.videoStreams)
+	for (const auto &i : program.videoStreams)
 		videoStreams.push_back(i.pid);
-	for (const auto i : program.audioStreams)
+	for (const auto &i : program.audioStreams)
 		audioStreams.push_back(i.pid);
-	for (const auto i : program.subtitleStreams)
+	for (const auto &i : program.subtitleStreams)
 		subtitleStreams.push_back(i.pid);
 	pcrPid = program.pcrPid;
 	pmtPid = program.pmtPid;
@@ -508,7 +508,7 @@ eDVBPMTParser::eStreamData::eStreamData(eDVBPMTParser::program &program)
 	adapterId = program.adapterId;
 	demuxId = program.demuxId;
 	serviceId = program.serviceId;
-	for (const auto it : program.caids)
+	for (const auto &it : program.caids)
 	{
 		caIds.push_back(it.caid);
 		ecmPids.push_back(it.capid);
