diff --git a/lib/python/Components/Language.py b/lib/python/Components/Language.py
index 7633a9f60..c41cc7fb6 100644
--- a/lib/python/Components/Language.py
+++ b/lib/python/Components/Language.py
@@ -8,7 +8,7 @@ from Tools.Directories import SCOPE_LANGUAGE, resolveFilename
 
 class Language:
 	def __init__(self):
-		gettext.install('enigma2', resolveFilename(SCOPE_LANGUAGE, ""), codeset="utf-8")
+		gettext.install('enigma2', resolveFilename(SCOPE_LANGUAGE, ""))
 		gettext.bindtextdomain("enigma2", resolveFilename(SCOPE_LANGUAGE))
 		gettext.textdomain("enigma2")
 		self.activeLanguage = 0
