diff --git a/lib/python/Components/About.py b/lib/python/Components/About.py
index 650143087..968e2ef84 100644
--- a/lib/python/Components/About.py
+++ b/lib/python/Components/About.py
@@ -4,6 +4,7 @@ import time
 import re
 from Tools.HardwareInfo import HardwareInfo
 from sys import maxsize, modules, version_info
+from subprocess import PIPE, Popen
 
 
 def getVersionString():
@@ -220,6 +221,15 @@ def getBoxUptime():
 	except:
 		return '-'
 
+def getOpenSSLVersion():
+	process = Popen(("/usr/bin/openssl", "version"), stdout=PIPE, stderr=PIPE, universal_newlines=True)
+	stdout, stderr = process.communicate()
+	if process.returncode == 0:
+		data = stdout.strip().split()
+		if len(data) > 1 and data[0] == "OpenSSL":
+			return data[1]
+	print("[About] Get OpenSSL version failed.")
+	return _("Unknown")
 
 # For modules that do "from About import about"
 about = modules[__name__]
diff --git a/lib/python/Screens/About.py b/lib/python/Screens/About.py
index f9ab0b4c8..d427625ef 100644
--- a/lib/python/Screens/About.py
+++ b/lib/python/Screens/About.py
@@ -78,6 +78,10 @@ class About(Screen):
 
 		AboutText += player + "\n"
 
+		AboutText += "\n"
+		AboutText += _("OpenSSL version: ") + about.getOpenSSLVersion() + "\n"
+
+		AboutText += "\n"
 		AboutText += _("Python version: ") + about.getPythonVersionString() + "\n"
 
 		AboutText += _("Enigma (re)starts: %d\n") % config.misc.startCounter.value
