diff --git a/lib/service/servicemp3.cpp b/lib/service/servicemp3.cpp
index 5c86fe3..53682ec 100644
--- a/lib/service/servicemp3.cpp
+++ b/lib/service/servicemp3.cpp
@@ -58,7 +58,7 @@ typedef enum
  * see: https://bugzilla.gnome.org/show_bug.cgi?id=619434
  * As a workaround, we run the subsink in sync=false mode
  */
-#define GSTREAMER_SUBTITLE_SYNC_MODE_BUG
+#undef GSTREAMER_SUBTITLE_SYNC_MODE_BUG
 /**/
 
 eServiceFactoryMP3::eServiceFactoryMP3()
