diff --git a/lib/python/Screens/About.py b/lib/python/Screens/About.py
index 9c231590d..841b00b99 100644
--- a/lib/python/Screens/About.py
+++ b/lib/python/Screens/About.py
@@ -220,11 +220,12 @@ class CommitInfo(Screen):
 		except:
 			branch = ""
 		branch_e2plugins = "?sha=python3"
+		branch_oe = "?sha=Kirkstone"
 
 		self.project = 0
 		self.projects = [
 			("https://api.github.com/repos/openpli/enigma2/commits" + branch, "Enigma2", API_GITHUB),
-			("https://api.github.com/repos/openpli/openpli-oe-core/commits" + branch, "Openpli Oe Core", API_GITHUB),
+			("https://api.github.com/repos/openpli/openpli-oe-core/commits" + branch_oe, "Openpli Oe Core", API_GITHUB),
 			("https://api.github.com/repos/openpli/enigma2-plugins/commits" + branch_e2plugins, "Enigma2 Plugins", API_GITHUB),
 			("https://api.github.com/repos/openpli/aio-grab/commits", "Aio Grab", API_GITHUB),
 			("https://api.github.com/repos/openpli/enigma2-plugin-extensions-epgimport/commits", "Plugin EPGImport", API_GITHUB),
