diff --git a/qthbbtv-v2/plugin/browser.py b/qthbbtv-v2/plugin/browser.py
index 7de761f..93756db 100644
--- a/qthbbtv-v2/plugin/browser.py
+++ b/qthbbtv-v2/plugin/browser.py
@@ -2,7 +2,7 @@ import os
 import struct
 from enigma import eConsoleAppContainer, getDesktop
 from Components.VolumeControl import VolumeControl
-import datasocket
+from . import datasocket
 
 
 class Browser:
diff --git a/qthbbtv-v2/plugin/hbbtv.py b/qthbbtv-v2/plugin/hbbtv.py
index 618df56..492e546 100644
--- a/qthbbtv-v2/plugin/hbbtv.py
+++ b/qthbbtv-v2/plugin/hbbtv.py
@@ -4,7 +4,7 @@ from Components.ServiceEventTracker import ServiceEventTracker
 from enigma import eTimer, iPlayableService
 import os
 from enigma import fbClass, eRCInput
-from browser import Browser
+from .browser import Browser
 from enigma import fbClass, eRCInput, eServiceReference
 import struct
 
diff --git a/qthbbtv-v2/plugin/plugin.py b/qthbbtv-v2/plugin/plugin.py
index c1f97fe..647b407 100644
--- a/qthbbtv-v2/plugin/plugin.py
+++ b/qthbbtv-v2/plugin/plugin.py
@@ -25,8 +25,7 @@ from Tools.Directories import fileExists, copyfile, pathExists, createDir
 from Components.ServicePosition import ServicePosition
 from Components.VolumeControl import VolumeControl
 import urllib
-import urllib2
-from hbbtv import HbbTVWindow
+from .hbbtv import HbbTVWindow
 
 
 
@@ -58,8 +57,8 @@ class HBBTVParser(Screen):
 		onid = info and info.getInfo(iServiceInformation.sONID) or -1
 		ssid = info and info.getInfo(iServiceInformation.sSID) or -1
 
-		print 'URL %s' % (startUrl)
-		print 'PMT %d TSID %d ONID %d SSID %d' % (pmt, tsid, onid, ssid)
+		print('URL %s' % (startUrl))
+		print('PMT %d TSID %d ONID %d SSID %d' % (pmt, tsid, onid, ssid))
 
 		ait = info.getAITApplications()
 
diff --git a/qthbbtv/plugin/browser.py b/qthbbtv/plugin/browser.py
index 7de761f..93756db 100644
--- a/qthbbtv/plugin/browser.py
+++ b/qthbbtv/plugin/browser.py
@@ -2,7 +2,7 @@ import os
 import struct
 from enigma import eConsoleAppContainer, getDesktop
 from Components.VolumeControl import VolumeControl
-import datasocket
+from . import datasocket
 
 
 class Browser:
diff --git a/qthbbtv/plugin/hbbtv.py b/qthbbtv/plugin/hbbtv.py
index 618df56..492e546 100644
--- a/qthbbtv/plugin/hbbtv.py
+++ b/qthbbtv/plugin/hbbtv.py
@@ -4,7 +4,7 @@ from Components.ServiceEventTracker import ServiceEventTracker
 from enigma import eTimer, iPlayableService
 import os
 from enigma import fbClass, eRCInput
-from browser import Browser
+from .browser import Browser
 from enigma import fbClass, eRCInput, eServiceReference
 import struct
 
diff --git a/qthbbtv/plugin/plugin.py b/qthbbtv/plugin/plugin.py
index c1f97fe..7ae5ba3 100644
--- a/qthbbtv/plugin/plugin.py
+++ b/qthbbtv/plugin/plugin.py
@@ -25,8 +25,7 @@ from Tools.Directories import fileExists, copyfile, pathExists, createDir
 from Components.ServicePosition import ServicePosition
 from Components.VolumeControl import VolumeControl
 import urllib
-import urllib2
-from hbbtv import HbbTVWindow
+from .hbbtv import HbbTVWindow
 
 
 
diff --git a/qtstalker-v2/plugin/browser.py b/qtstalker-v2/plugin/browser.py
index 7e87b3d..00f6614 100644
--- a/qtstalker-v2/plugin/browser.py
+++ b/qtstalker-v2/plugin/browser.py
@@ -2,7 +2,7 @@ import os
 import struct
 from enigma import eConsoleAppContainer, getDesktop
 from Components.VolumeControl import VolumeControl
-import datasocket
+from . import datasocket
 
 
 class Browser:
diff --git a/qtstalker-v2/plugin/plugin.py b/qtstalker-v2/plugin/plugin.py
index 85aa909..b054281 100644
--- a/qtstalker-v2/plugin/plugin.py
+++ b/qtstalker-v2/plugin/plugin.py
@@ -9,7 +9,7 @@ from enigma import iServiceInformation, eTimer, eConsoleAppContainer
 from Plugins.Plugin import PluginDescriptor
 from Screens.Screen import Screen
 from Screens.MessageBox import MessageBox
-from stalker import StalkerTVWindow
+from .stalker import StalkerTVWindow
 import os
 import netifaces
 import datetime
diff --git a/qtstalker-v2/plugin/stalker.py b/qtstalker-v2/plugin/stalker.py
index 97741e3..f770d77 100644
--- a/qtstalker-v2/plugin/stalker.py
+++ b/qtstalker-v2/plugin/stalker.py
@@ -2,7 +2,7 @@ from Screens.Screen import Screen
 from Components.ActionMap import ActionMap
 from Components.ServiceEventTracker import ServiceEventTracker
 from enigma import fbClass, eRCInput, gMainDC, getDesktop, eSize, eServiceReference, eTimer, iPlayableService, eDVBVolumecontrol
-from browser import Browser
+from .browser import Browser
 from Components.config import config
 import struct
 
diff --git a/qtstalker/plugin/browser.py b/qtstalker/plugin/browser.py
index 7e87b3d..00f6614 100644
--- a/qtstalker/plugin/browser.py
+++ b/qtstalker/plugin/browser.py
@@ -2,7 +2,7 @@ import os
 import struct
 from enigma import eConsoleAppContainer, getDesktop
 from Components.VolumeControl import VolumeControl
-import datasocket
+from . import datasocket
 
 
 class Browser:
diff --git a/qtstalker/plugin/plugin.py b/qtstalker/plugin/plugin.py
index 80c6335..403b553 100644
--- a/qtstalker/plugin/plugin.py
+++ b/qtstalker/plugin/plugin.py
@@ -9,7 +9,7 @@ from enigma import iServiceInformation, eTimer, eConsoleAppContainer
 from Plugins.Plugin import PluginDescriptor
 from Screens.Screen import Screen
 from Screens.MessageBox import MessageBox
-from stalker import StalkerTVWindow
+from .stalker import StalkerTVWindow
 import os
 import netifaces
 import datetime
diff --git a/qtstalker/plugin/stalker.py b/qtstalker/plugin/stalker.py
index ec584dd..67f1d13 100644
--- a/qtstalker/plugin/stalker.py
+++ b/qtstalker/plugin/stalker.py
@@ -2,7 +2,7 @@ from Screens.Screen import Screen
 from Components.ActionMap import ActionMap
 from Components.ServiceEventTracker import ServiceEventTracker
 from enigma import fbClass, eRCInput, gMainDC, getDesktop, eSize, eServiceReference, eTimer, iPlayableService, eDVBVolumecontrol
-from browser import Browser
+from .browser import Browser
 from Components.config import config
 import struct
 
