diff --git a/os_dep/linux/os_intfs.c b/os_dep/linux/os_intfs.c
index 46385c5..30cec67 100644
--- a/os_dep/linux/os_intfs.c
+++ b/os_dep/linux/os_intfs.c
@@ -698,12 +698,18 @@ unsigned int rtw_classify8021d(struct sk_buff *skb)
 
  
 static u16 rtw_select_queue(struct net_device *dev, struct sk_buff *skb
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0) 	
-				, void *accel_priv
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0) 
-				, select_queue_fallback_t fallback
-#endif
-
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,19,0))
+	, struct net_device *sb_dev
+	#if (LINUX_VERSION_CODE < KERNEL_VERSION(5,2,0))
+	, select_queue_fallback_t fallback
+	#endif
+#else
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 13, 0)
+	, void *accel_priv
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+	, select_queue_fallback_t fallback
+	#endif
+	#endif
 #endif
 )
 {
diff --git a/platform/platform_ops.c b/platform/platform_ops.c
index 1583e4c..b47f24c 100644
--- a/platform/platform_ops.c
+++ b/platform/platform_ops.c
@@ -20,8 +20,11 @@
 #ifndef CONFIG_PLATFORM_OPS
 #include <linux/time.h>
 #include <linux/delay.h>
+#include <linux/version.h>
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5,2,0))
 extern void sdio_reinit(void);
 extern void extern_wifi_set_enable(int is_on);
+#endif
 /*
  * Return:
  *	0:	power on successfully
@@ -30,12 +33,14 @@ extern void extern_wifi_set_enable(int is_on);
 int platform_wifi_power_on(void)
 {
 	int ret = 0;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5,2,0))
 
 	extern_wifi_set_enable(0);
 	msleep(500);
 	extern_wifi_set_enable(1);
 	msleep(500);
 	sdio_reinit();
+#endif
 	return ret;
 }
 
