diff --git a/configure.ac b/configure.ac
index 48e3a11..463bfd6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,6 +57,7 @@ AC_LINK_IFELSE(
     SWIG_VOID(RESULT) subServices(ePtr<iSubserviceList> &SWIG_OUTPUT){};
     SWIG_VOID(RESULT) frontendInfo(ePtr<iFrontendInformation> &SWIG_OUTPUT){};
     SWIG_VOID(RESULT) timeshift(ePtr<iTimeshiftService> &SWIG_OUTPUT){};
+    SWIG_VOID(RESULT) tap(ePtr<iTapService> &ptr) { ptr = nullptr; return -1; };
     SWIG_VOID(RESULT) cueSheet(ePtr<iCueSheet> &SWIG_OUTPUT){};
     SWIG_VOID(RESULT) subtitle(ePtr<iSubtitleOutput> &SWIG_OUTPUT){};
     SWIG_VOID(RESULT) audioDelay(ePtr<iAudioDelay> &SWIG_OUTPUT){};
diff --git a/src/serviceapp/serviceapp.h b/src/serviceapp/serviceapp.h
index 7bb0cc6..3fe9986 100644
--- a/src/serviceapp/serviceapp.h
+++ b/src/serviceapp/serviceapp.h
@@ -124,6 +124,7 @@ public:
 	RESULT subServices(ePtr<iSubserviceList> &ptr){ ptr=this; return 0;};
 	RESULT frontendInfo(ePtr<iFrontendInformation> &ptr){ ptr=0; return -1;};
 	RESULT timeshift(ePtr<iTimeshiftService> &ptr){ ptr=0; return -1;};
+	RESULT tap(ePtr<iTapService> &ptr) { ptr = nullptr; return -1; };
 	RESULT cueSheet(ePtr<iCueSheet> &ptr){ ptr=0; return -1;};
 	RESULT subtitle(ePtr<iSubtitleOutput> &ptr){ ptr=this; return 0;};
 	RESULT audioDelay(ePtr<iAudioDelay> &ptr){ ptr=0; return -1;};
