diff --git a/gst-libs/ext/libav/libavcodec/huffyuv.c b/gst-libs/ext/libav/libavcodec/huffyuv.c
index a399142..a298cb5 100644
--- a/gst-libs/ext/libav/libavcodec/huffyuv.c
+++ b/gst-libs/ext/libav/libavcodec/huffyuv.c
@@ -34,6 +34,10 @@
 #include "dsputil.h"
 #include "thread.h"
 
+#ifndef AV_PIX_FMT_YUV422P
+#define AV_PIX_FMT_YUV422P 4
+#endif
+
 #define VLC_BITS 11
 
 #if HAVE_BIGENDIAN
@@ -517,6 +521,10 @@ s->bgr32=1;
         assert(0);
     }
 
+    if (s->predictor == MEDIAN && avctx->pix_fmt == AV_PIX_FMT_YUV422P && avctx->width%4) {
+        av_log(avctx, AV_LOG_ERROR, "width must be a multiple of 4 this colorspace and predictor\n");
+        return AVERROR_INVALIDDATA;
+    }
     alloc_temp(s);
 
 //    av_log(NULL, AV_LOG_DEBUG, "pred:%d bpp:%d hbpp:%d il:%d\n", s->predictor, s->bitstream_bpp, avctx->bits_per_coded_sample, s->interlaced);
