From 17fca82124417bf030f883b591bf4b8cb9d04706 Mon Sep 17 00:00:00 2001
From: ims <ims21@users.sourceforge.net>
Date: Wed, 24 Nov 2021 19:57:16 +0100
Subject: [PATCH] support kernel 5.1 to 5.9

---
 os_dep/osdep_service.c | 39 +++++++++++++++++++++++++++------------
 1 file changed, 27 insertions(+), 12 deletions(-)

diff --git a/os_dep/osdep_service.c b/os_dep/osdep_service.c
index fea029e..2aefc54 100644
--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -2202,9 +2202,14 @@ static int isFileReadable(const char *path, u32 *sz)
 	if (IS_ERR(fp))
 		ret = PTR_ERR(fp);
 	else {
-        #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
+	#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 		oldfs = get_fs();
-        #endif
+		#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
+		set_fs(KERNEL_DS);
+		#else
+		set_fs(get_ds());
+		#endif
+	#endif
 
 		if (1 != readFile(fp, &buf, 1))
 			ret = PTR_ERR(fp);
@@ -2219,7 +2224,7 @@ static int isFileReadable(const char *path, u32 *sz)
 
 		#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 		set_fs(oldfs);
-        #endif
+		#endif
 		filp_close(fp, NULL);
 	}
 	return ret;
@@ -2245,13 +2250,18 @@ static int retriveFromFile(const char *path, u8 *buf, u32 sz)
 		if (0 == ret) {
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
-            #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
+		#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			oldfs = get_fs();
-            #endif
+			#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
+			set_fs(KERNEL_DS);
+			#else
+			set_fs(get_ds());
+			#endif
+		#endif
 			ret = readFile(fp, buf, sz);
-            #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
+		#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			set_fs(oldfs);
-            #endif
+		#endif
 			closeFile(fp);
 
 			RTW_INFO("%s readFile, ret:%d\n", __FUNCTION__, ret);
@@ -2285,13 +2295,18 @@ static int storeToFile(const char *path, u8 *buf, u32 sz)
 		if (0 == ret) {
 			RTW_INFO("%s openFile path:%s fp=%p\n", __FUNCTION__, path , fp);
 
-            #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
+		#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 			oldfs = get_fs();
-            #endif
+			#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0))
+			set_fs(KERNEL_DS);
+			#else
+			set_fs(get_ds());
+			#endif
+		#endif
 			ret = writeFile(fp, buf, sz);
-            #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
-            set_fs(oldfs);
-            #endif
+		#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
+			set_fs(oldfs);
+		#endif
 			closeFile(fp);
 
 			RTW_INFO("%s writeFile, ret:%d\n", __FUNCTION__, ret);
-- 
2.17.1

