diff -ur node-v12.21.0/deps/v8/src/inspector/v8-string-conversions.h node-v12.21.0/deps/v8/src/inspector/v8-string-conversions.h
--- node-v12.21.0/deps/v8/src/inspector/v8-string-conversions.h	2021-02-23 03:58:06.000000000 +0100
+++ node-v12.21.0/deps/v8/src/inspector/v8-string-conversions.h	2023-08-27 15:09:24.367444103 +0200
@@ -6,6 +6,7 @@
 #define V8_INSPECTOR_V8_STRING_CONVERSIONS_H_
 
 #include <string>
+#include <cstdint>
 
 // Conversion routines between UT8 and UTF16, used by string-16.{h,cc}. You may
 // want to use string-16.h directly rather than these.
