diff --git a/plugin/rbbmanager.py b/plugin/rbbmanager.py
index 4b15561..28067fd 100644
--- a/plugin/rbbmanager.py
+++ b/plugin/rbbmanager.py
@@ -131,13 +131,13 @@ class refreshBouquetRbbManager(Screen):
 				os.rename(path, newpath)
 				self.reloadList()
 				return
-			except OSError, e:
+			except OSError as e:
 				print("Error %s:" % e.errno, e)
 				if e.errno == 17:
 					msg = _("The path %s already exists.") % name
 				else:
 					msg = _("Error") + '\n' + str(e)
-			except Exception, e:
+			except Exception as e:
 				import traceback
 				print("[ML] Unexpected error:", e)
 				traceback.print_exc()
diff --git a/setup_translate.py b/setup_translate.py
index 29e8221..7f8e807 100644
--- a/setup_translate.py
+++ b/setup_translate.py
@@ -21,9 +21,9 @@ class build_trans(cmd.Command):
 					if f.endswith('.po'):
 						src = os.path.join(lc, f)
 						dest = os.path.join(lc, f[:-2] + 'mo')
-						print "Language compile %s -> %s" % (src, dest)
+						print("Language compile %s -> %s" % (src, dest))
 						if os.system("msgfmt '%s' -o '%s'" % (src, dest)) != 0:
-							raise Exception, "Failed to compile: " + src
+							raise Exception("Failed to compile: " + src)
 
 class build(_build):
 	sub_commands = _build.sub_commands + [('build_trans', None)]
