diff --git a/src/streamlink/compat.py b/src/streamlink/compat.py
index 4fee3aa8..975089e0 100644
--- a/src/streamlink/compat.py
+++ b/src/streamlink/compat.py
@@ -5,9 +5,6 @@ import sys
 import warnings
 from typing import Any, Callable, Dict, Optional, Tuple
 
-# import exceptiongroup, so it can monkeypatch ExceptionGroup logic on <=py311
-import exceptiongroup  # noqa: F401
-
 from streamlink.exceptions import StreamlinkDeprecationWarning
 
 
