Index: qemu-2.8.0/linux-user/syscall.c
===================================================================
--- qemu-2.8.0.orig/linux-user/syscall.c
+++ qemu-2.8.0/linux-user/syscall.c
@@ -252,16 +252,9 @@
 #define TARGET_NR__llseek TARGET_NR_llseek
 #endif
 
-#ifdef __NR_gettid
 #define __NR_sys_gettid __NR_gettid
 _syscall0(int, sys_gettid)
-#else
-/* This is a replacement for the host gettid() and must return a host
-   errno. */
-static int gettid(void) {
-    return -ENOSYS;
-}
-#endif
+
 #if defined(TARGET_NR_getdents) && defined(__NR_getdents)
 _syscall3(int, sys_getdents, uint, fd, struct linux_dirent *, dirp, uint, count);
 #endif
