diff --git a/plugin/controllers/epg.py b/plugin/controllers/epg.py
index 5f15e3f6..19079371 100644
--- a/plugin/controllers/epg.py
+++ b/plugin/controllers/epg.py
@@ -27,7 +27,7 @@ import json
 from enigma import eEPGCache, eServiceReference
 from Components.config import config
 
-DEBUG_ENABLED = config.OpenWebif.verbose_debug_enabled.value
+DEBUG_ENABLED = False
 CASE_SENSITIVE_QUERY = 0
 CASE_INSENSITIVE_QUERY = 1
 REGEX_QUERY =2
