diff --git a/pythonwifi/iwlibs.py b/pythonwifi/iwlibs.py
index 5fedede..5e3789b 100644
--- a/pythonwifi/iwlibs.py
+++ b/pythonwifi/iwlibs.py
@@ -1057,8 +1057,7 @@ class Iwstruct(object):
         if (sys.version_info[0] == 2):            
             ifreq = array.array('c', ifname + '\0'*buff)
         else:
-            var_bytes = bytes(ifname, 'unicode-escape')        
-            ifreq = array.array('B', var_bytes + b'\0'*buff)
+            ifreq = array.array('B', ifname + b'\0'*buff)
         # put some additional data behind the interface name
         if data is not None:
             ifreq.extend(data)
