diff --git a/pythonwifi/iwlibs.py b/pythonwifi/iwlibs.py
index 41f848e..db255b2 100644
--- a/pythonwifi/iwlibs.py
+++ b/pythonwifi/iwlibs.py
@@ -1437,7 +1437,7 @@ class Iwscan(object):
     def __len__(self):
         return len(self.aplist)
 
-    def next(self):
+    def __next__(self):
         self.index = self.index + 1
         if self.index > len(self.aplist) - 1:
             raise StopIteration
