diff -Naur gst-plugins-base.orig/gst-libs/gst/allocators/Makefile.am gst-plugins-base/gst-libs/gst/allocators/Makefile.am
--- a/gst-libs/gst/allocators/Makefile.am	2018-03-21 13:19:46.841465958 +0100
+++ b/gst-libs/gst/allocators/Makefile.am	2018-03-21 13:20:07.244954514 +0100
@@ -42,7 +42,7 @@
 		--add-include-path=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-@GST_API_VERSION@` \
 		--library=libgstallocators-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg-export gstreamer-allocators-@GST_API_VERSION@ \
 		--output $@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/app/Makefile.am gst-plugins-base/gst-libs/gst/app/Makefile.am
--- a/gst-libs/gst/app/Makefile.am	2018-03-21 13:19:46.841465958 +0100
+++ b/gst-libs/gst/app/Makefile.am	2018-03-21 13:20:07.244954514 +0100
@@ -56,7 +56,7 @@
 		--library=libgstapp-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-app-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/audio/Makefile.am gst-plugins-base/gst-libs/gst/audio/Makefile.am
--- a/gst-libs/gst/audio/Makefile.am	2018-03-21 13:19:46.842465933 +0100
+++ b/gst-libs/gst/audio/Makefile.am	2018-03-21 13:20:07.244954514 +0100
@@ -184,7 +184,7 @@
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
 		--include=GstTag-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-audio-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/gl/Makefile.am gst-plugins-base/gst-libs/gst/gl/Makefile.am
--- a/gst-libs/gst/gl/Makefile.am	2018-03-21 13:19:46.850465732 +0100
+++ b/gst-libs/gst/gl/Makefile.am	2018-03-21 13:22:34.844254721 +0100
@@ -192,7 +192,7 @@
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
 		--include=GstVideo-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg gstreamer-video-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/pbutils/Makefile.am gst-plugins-base/gst-libs/gst/pbutils/Makefile.am
--- a/gst-libs/gst/pbutils/Makefile.am	2018-03-21 13:19:46.860465482 +0100
+++ b/gst-libs/gst/pbutils/Makefile.am	2018-03-21 13:20:07.245954489 +0100
@@ -103,7 +103,7 @@
 		--include=GstTag-@GST_API_VERSION@ \
 		--include=GstVideo-@GST_API_VERSION@ \
 		--include=GstAudio-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-tag-@GST_API_VERSION@ \
 		--pkg gstreamer-video-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/riff/Makefile.am gst-plugins-base/gst-libs/gst/riff/Makefile.am
--- a/gst-libs/gst/riff/Makefile.am	2018-03-21 13:19:46.863465406 +0100
+++ b/gst-libs/gst/riff/Makefile.am	2018-03-21 13:20:07.245954489 +0100
@@ -49,7 +49,7 @@
 #		--include=GstAudio-@GST_API_VERSION@ \
 #		--include=GstTag-@GST_API_VERSION@ \
 #		--include=Gst-@GST_API_VERSION@ \
-#		--libtool="$(top_builddir)/libtool" \
+#		--libtool="$(LIBTOOL)" \
 #		--pkg gstreamer-@GST_API_VERSION@ \
 #		--pkg gstreamer-tag-@GST_API_VERSION@ \
 #		--pkg gstreamer-audio-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/rtp/Makefile.am gst-plugins-base/gst-libs/gst/rtp/Makefile.am
--- a/gst-libs/gst/rtp/Makefile.am	2018-03-21 13:19:46.864465381 +0100
+++ b/gst-libs/gst/rtp/Makefile.am	2018-03-21 13:20:07.245954489 +0100
@@ -69,7 +69,7 @@
 		--library=libgstrtp-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-rtp-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/rtsp/Makefile.am gst-plugins-base/gst-libs/gst/rtsp/Makefile.am
--- a/gst-libs/gst/rtsp/Makefile.am	2018-03-21 13:19:46.866465331 +0100
+++ b/gst-libs/gst/rtsp/Makefile.am	2018-03-21 13:20:07.245954489 +0100
@@ -76,7 +76,7 @@
 		--include=Gio-2.0 \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstSdp-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gio-2.0 \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-sdp-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/sdp/Makefile.am gst-plugins-base/gst-libs/gst/sdp/Makefile.am
--- a/gst-libs/gst/sdp/Makefile.am	2018-03-21 13:19:46.867465306 +0100
+++ b/gst-libs/gst/sdp/Makefile.am	2018-03-21 13:20:07.246954464 +0100
@@ -34,7 +34,7 @@
 		--add-include-path=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-@GST_API_VERSION@` \
 		--library=libgstsdp-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg-export gstreamer-sdp-@GST_API_VERSION@ \
 		--output $@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/tag/Makefile.am gst-plugins-base/gst-libs/gst/tag/Makefile.am
--- a/gst-libs/gst/tag/Makefile.am	2018-03-21 13:19:46.868465281 +0100
+++ b/gst-libs/gst/tag/Makefile.am	2018-03-21 13:20:07.246954464 +0100
@@ -66,7 +66,7 @@
 		--library=libgsttag-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-tag-@GST_API_VERSION@ \
diff -Naur gst-plugins-base.orig/gst-libs/gst/video/Makefile.am gst-plugins-base/gst-libs/gst/video/Makefile.am
--- a/gst-libs/gst/video/Makefile.am	2018-03-21 13:19:46.872465181 +0100
+++ b/gst-libs/gst/video/Makefile.am	2018-03-21 13:20:07.246954464 +0100
@@ -124,7 +124,7 @@
 		--library=libgstvideo-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
-		--libtool="$(top_builddir)/libtool" \
+		--libtool="$(LIBTOOL)" \
 		--pkg gstreamer-@GST_API_VERSION@ \
 		--pkg gstreamer-base-@GST_API_VERSION@ \
 		--pkg-export gstreamer-video-@GST_API_VERSION@ \
