From 979c5cb85c4f3e59dcc44c1e6c75315dee7dfbc2 Mon Sep 17 00:00:00 2001
From: Maros Ondrasek <mx3ldev@gmail.com>
Date: Thu, 6 Nov 2014 14:03:57 +0100
Subject: [PATCH] riff-media: added fourcc to all mpeg4 video caps

---
 gst-libs/gst/riff/riff-media.c | 32 ++++
 1 file changed, 4 insertions(+)

diff --git a/gst-libs/gst/riff/riff-media.c b/gst-libs/gst/riff/riff-media.c
index 2d6aa55d8..7fa70b963 100644
--- a/gst-libs/gst/riff/riff-media.c
+++ b/gst-libs/gst/riff/riff-media.c
@@ -253,6 +253,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "systemstream", G_TYPE_BOOLEAN, FALSE,
           "mpegversion", G_TYPE_INT, 1, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("MPEG-1 video");
       break;
@@ -265,6 +269,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "systemstream", G_TYPE_BOOLEAN, FALSE,
           "mpegversion", G_TYPE_INT, 2, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("MPEG-2 video");
       break;
@@ -273,6 +281,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "systemstream", G_TYPE_BOOLEAN, FALSE,
           "mpegversion", G_TYPE_INT, 2, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("Lead MPEG-2 video");
       break;
@@ -372,6 +384,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "mpegversion", G_TYPE_INT, 4,
           "systemstream", G_TYPE_BOOLEAN, FALSE, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("Samsung MPEG-4");
       break;
@@ -380,6 +396,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "mpegversion", G_TYPE_INT, 4,
           "systemstream", G_TYPE_BOOLEAN, FALSE, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("Divio MPEG-4");
       break;
@@ -459,6 +479,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "mpegversion", G_TYPE_INT, 4,
           "systemstream", G_TYPE_BOOLEAN, FALSE, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("Microsoft ISO MPEG-4 1.1");
       break;
@@ -469,6 +493,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "mpegversion", G_TYPE_INT, 4,
           "systemstream", G_TYPE_BOOLEAN, FALSE, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("FFmpeg MPEG-4");
       break;
@@ -497,6 +525,10 @@ gst_riff_create_video_caps (guint32 codec_fcc,
       caps = gst_caps_new_simple ("video/mpeg",
           "mpegversion", G_TYPE_INT, 4,
           "systemstream", G_TYPE_BOOLEAN, FALSE, NULL);
+      {gchar *fstr = g_strdup_printf ("%" GST_FOURCC_FORMAT,
+          GST_FOURCC_ARGS (codec_fcc));
+      gst_caps_set_simple (caps, "fourcc", G_TYPE_STRING, fstr, NULL);
+      g_free (fstr);}
       if (codec_name)
         *codec_name = g_strdup ("MPEG-4");
       break;
-- 
2.1.4

