Index: glib-2.34.2/tests/threadpool-test.c
===================================================================
--- glib-2.34.2.orig/tests/threadpool-test.c	2012-11-19 19:21:23.007594309 +0100
+++ glib-2.34.2/tests/threadpool-test.c	2012-11-19 19:21:45.395598384 +0100
@@ -5,7 +5,7 @@
 
 #include <glib.h>
 
-/* #define DEBUG 1 */
+#define DEBUG 1
 
 #ifdef DEBUG
 # define DEBUG_MSG(args) g_printerr args ; g_printerr ("\n");
@@ -113,10 +113,10 @@
        if (g_thread_pool_get_num_unused_threads () == 0)
          break;
 
-       DEBUG_MSG (("[unused] waiting ONE second for threads to die"));
+       DEBUG_MSG (("[unused] waiting FIVE second for threads to die"));
 
        /* Some time for threads to die. */
-       g_usleep (G_USEC_PER_SEC);
+       g_usleep (5 * G_USEC_PER_SEC);
      }
 
    DEBUG_MSG (("[unused] stopped idle threads, %d remain",
