--- glib-2.34.2.orig/Makefile.am     2012-11-19 16:11:19.499596880 +0100
+++ glib-2.34.2/Makefile.am 2012-11-19 16:04:36.859596461 +0100
@@ -6,7 +6,7 @@

 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}

-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 if BUILD_MODULAR_TESTS
 SUBDIRS += tests

