diff --git a/lib/python/Plugins/Extensions/HbbTV/plugin.py b/lib/python/Plugins/Extensions/HbbTV/plugin.py
index 73e2074..35fa055 100644
--- a/lib/python/Plugins/Extensions/HbbTV/plugin.py
+++ b/lib/python/Plugins/Extensions/HbbTV/plugin.py
@@ -183,6 +183,7 @@ class OpCodeSet:
 			,"OP_HBBTV_TITLE"		: 0x0006
 			,"OP_HBBTV_RETRY_OPEN_URL"	: 0x0009
 			,"OP_HBBTV_CHANGE_CHANNEL"	: 0x000A
+			,"OP_HBBTV_ABORT_SIG"		: 0x000B
 			,"OP_OIPF_GET_CHANNEL_INFO_URL"	: 0x0101
 			,"OP_OIPF_GET_CHANNEL_INFO_AIT" : 0x0102
 			,"OP_OIPF_GET_CHANNEL_INFO_LIST": 0x0103
@@ -223,6 +224,7 @@ class OpCodeSet:
 			,0x0006 : "OP_HBBTV_TITLE"
 			,0x0009 : "OP_HBBTV_RETRY_OPEN_URL"
 			,0x000A : "OP_HBBTV_CHANGE_CHANNEL"
+			,0x000B : "OP_HBBTV_ABORT_SIG"
 			,0x0101 : "OP_OIPF_GET_CHANNEL_INFO_URL"
 			,0x0102 : "OP_OIPF_GET_CHANNEL_INFO_AIT"
 			,0x0103 : "OP_OIPF_GET_CHANNEL_INFO_LIST"
@@ -450,6 +452,7 @@ class HandlerHbbTV(Handler):
 			,0x0006 : self._cb_handleSetPageTitle
 			,0x0009 : self._cb_handleHbbTVRetryOpen
 			,0x000A : self._cb_handleHbbTVChangeChannel
+			,0x000B : self._cb_handleHbbTVAbortSig
 			,0x0101 : self._cb_handleGetChannelInfoForUrl
 			,0x0102 : self._cb_handleGetChannelInfoForAIT
 			,0x0103 : self._cb_handleGetChannelInfoList
@@ -709,6 +712,11 @@ class HandlerHbbTV(Handler):
 						self._on_set_title_cb.remove(x)
 		return (0, "OK")
 
+	def _cb_handleHbbTVAbortSig(self, opcode, data):
+		self._cb_handleCloseHbbTVBrowser(opcode, data)
+		global __gval__
+		return (0, struct.pack('!II', __gval__.resX, __gval__.resY));
+
 	def _cb_handleCloseHbbTVBrowser(self, opcode, data):
 		self._timer_retry_open.stop()
 		try:
@@ -743,6 +751,12 @@ class HandlerHbbTV(Handler):
 		vodUri = data[hl:hl+uriLength]
 		self._handle_dump(self._cb_handleVODPlayerURI, opcode, vodUri)
 		self._vod_uri = vodUri
+		try:
+			if vodUri.find('/zdf/') >= 0:
+				tmpUri = MMSStreamURL().getLocationData(vodUri).strip()
+				if not strIsEmpty(tmpUri):
+					self._vod_uri = tmpUri
+		except: pass
 		return (0, "OK")
 
 	def doStop(self, restoreBeforeService=True, needStop=True):
@@ -866,7 +880,7 @@ class HbbTVWindow(Screen, InfoBarNotifications):
 			self._vod_length = length
 			if position == -1 and length == -1:
 				raise Exception("Can't get play status")
-			#print getTimeString(position), "/", getTimeString(length)
+			#print position, "/", length, " - ", getTimeString(position), "/", getTimeString(length)
 			self._ssm.setStatus(position, length, 1)
 		except Exception, ErrMsg:
 			print ErrMsg
@@ -1073,6 +1087,7 @@ class HbbTVHelper(Screen, InfoBarNotifications):
 		global HBBTVAPP_PATH
 		try:	os.system('%s/launcher stop'%(HBBTVAPP_PATH))
 		except: pass
+		return True
 
 	def _restart_opera(self):
 		global HBBTVAPP_PATH
