diff --git a/lib/python/Plugins/Extensions/HbbTV/plugin.py b/lib/python/Plugins/Extensions/HbbTV/plugin.py
--- a/lib/python/Plugins/Extensions/HbbTV/plugin.py	2014-01-06 14:56:45.001854323 +0100
+++ b/lib/python/Plugins/Extensions/HbbTV/plugin.py	2014-01-06 15:02:42.317916489 +0100
@@ -2339,7 +2339,7 @@
     def makeConfigList(self):
 	self.menulist = []
 	entryUri = getConfigListEntry(_("YouTube TV URL"), config.plugins.youtubetv.uri)
-	entryShowHelp = getConfigListEntry(_("Do not show YouTube TV Stater again"), config.plugins.youtubetv.showhelp)
+	entryShowHelp = getConfigListEntry(_("Do not show YouTube TV Starter again"), config.plugins.youtubetv.showhelp)
 	self.menulist.append(entryUri)
 	self.menulist.append(entryShowHelp)
 
@@ -2431,9 +2431,7 @@
 	session.open(YoutubeTVSettings)
 
 def start_menu_main(menuid, **kwargs):
-	if menuid == "mainmenu":
-		return [(_("YouTube TV"), showYoutubeTV, "youtube_tv", 46)]
-	return []
+	return [(_("YouTube TV"), showYoutubeTV, "youtube_tv", 46)]
 
 def plugin_start_main(session, **kwargs):
 	#session.open(OperaBrowser)
@@ -2462,7 +2460,7 @@
 def Plugins(path, **kwargs):
 	l = []
 	l.append(PluginDescriptor(where=PluginDescriptor.WHERE_AUTOSTART, fnc=auto_start_main))
-	l.append(PluginDescriptor(name=_("YouTube TV"), where=PluginDescriptor.WHERE_MENU, fnc=start_menu_main))
+	l.append(PluginDescriptor(name=_("YouTube TV"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, fnc=start_menu_main))
 	l.append(PluginDescriptor(name=_("YouTube TV Settings"), where=PluginDescriptor.WHERE_PLUGINMENU, fnc=youtube_setting_main))
 	l.append(PluginDescriptor(where=PluginDescriptor.WHERE_SESSIONSTART, needsRestart=True, fnc=session_start_main, weight=-10))
 	l.append(PluginDescriptor(name=_("HbbTV Applications"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, needsRestart=True, fnc=plugin_extension_start_application))
