diff --git a/archival/libarchive/data_extract_all.c b/archival/libarchive/data_extract_all.c
index 3142405..368108d 100644
--- a/archival/libarchive/data_extract_all.c
+++ b/archival/libarchive/data_extract_all.c
@@ -132,6 +132,7 @@ void FAST_FUNC data_extract_all(archive_handle_t *archive_handle)
 
 	/* Create the filesystem entry */
 	switch (file_header->mode & S_IFMT) {
+	case 0:
 	case S_IFREG: {
 		/* Regular file */
 		char *dst_nameN;
@@ -207,7 +208,7 @@ void FAST_FUNC data_extract_all(archive_handle_t *archive_handle)
 		}
 		break;
 	default:
-		bb_simple_error_msg_and_die("unrecognized file type");
+		bb_perror_msg_and_die("unrecognized file type, mode is 0x%x", file_header->mode & S_IFMT);
 	}
 
 	if (!S_ISLNK(file_header->mode)) {
