diff --git a/libopkg/opkg_install.c b/libopkg/opkg_install.c
index f778348..4dab4d5 100644
--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -431,6 +431,7 @@ static int check_data_file_clashes(pkg_t * pkg, pkg_t * old_pkg)
     file_info_t *file_info;
     char *filename;
     int clashes = 0;
+    struct stat target_stat;
 
     files_list = pkg_get_installed_files(pkg);
     if (files_list == NULL)
@@ -443,10 +444,20 @@ static int check_data_file_clashes(pkg_t * pkg, pkg_t * old_pkg)
         if (file_exists(filename)) {
             pkg_t *owner;
             pkg_t *obs;
-            int existing_is_dir = file_is_dir(filename);
+            int existing_is_dir = 0;
+            int existing_is_symlink = 0;
 
+            if(!xlstat(filename, &target_stat))
+                existing_is_symlink = S_ISLNK(target_stat.st_mode);
+
+            if(!stat(filename, &target_stat))
+                existing_is_dir = S_ISDIR(target_stat.st_mode);
+
+            /* OK if both are symlinks, either to a file or a directory */
+            if (existing_is_symlink && S_ISLNK(file_info->mode))
+                continue;
             /* OK if both the existing file and new file are directories. */
-            if (existing_is_dir && S_ISDIR(file_info->mode)) {
+            else if (existing_is_dir && S_ISDIR(file_info->mode)) {
                 continue;
             } else if (existing_is_dir || S_ISDIR(file_info->mode)) {
                 /* OK if existing file is a symlink to a directory and the new
@@ -491,7 +502,6 @@ static int check_data_file_clashes(pkg_t * pkg, pkg_t * old_pkg)
             if (S_ISLNK(file_info->mode) && file_is_symlink(filename)) {
                 char *link_target;
                 int r, target_is_same_directory = 0;
-                struct stat target_stat;
 
                 link_target = file_readlink_alloc(filename);
                 r = strcmp(link_target, file_info->link_target);
-- 
2.25.1

