diff --git a/xbmc/platform/linux/input/LibInputKeyboard.cpp b/xbmc/platform/linux/input/LibInputKeyboard.cpp
index 657588adce..ce48fa4c37 100644
--- a/xbmc/platform/linux/input/LibInputKeyboard.cpp
+++ b/xbmc/platform/linux/input/LibInputKeyboard.cpp
@@ -55,7 +55,7 @@ static const std::map<xkb_keysym_t, XBMCKey> xkbMap =
   // XKB_KEY_XF86Favorites could be XBMCK_BROWSER_FAVORITES or XBMCK_FAVORITES,
   // XBMCK_FAVORITES was chosen here because it is more general
   { XKB_KEY_XF86HomePage, XBMCK_BROWSER_HOME },
-  { XKB_KEY_XF86AudioMute, XBMCK_VOLUME_MUTE },
+  { XKB_KEY_XF86AudioMute, XBMCK_RETURN },
   { XKB_KEY_XF86AudioLowerVolume, XBMCK_VOLUME_DOWN },
   { XKB_KEY_XF86AudioRaiseVolume, XBMCK_VOLUME_UP },
   { XKB_KEY_XF86AudioNext, XBMCK_MEDIA_NEXT_TRACK },
