diff --git a/cmake/treedata/common/cores.txt b/cmake/treedata/common/cores.txt
index 5e5c6cb..cbde630 100644
--- a/cmake/treedata/common/cores.txt
+++ b/cmake/treedata/common/cores.txt
@@ -6,3 +6,4 @@ xbmc/cores/DllLoader/exports/util      cores/dll-loader/exports/util
 xbmc/cores/ExternalPlayer              cores/externalplayer
 xbmc/cores/paplayer                    cores/paplayer
 xbmc/cores/playercorefactory           cores/playercorefactory
+xbmc/cores/GstPlayer                   cores/gstplayer
diff --git a/xbmc/cores/playercorefactory/PlayerCoreConfig.h b/xbmc/cores/playercorefactory/PlayerCoreConfig.h
index 488947e90ad..37e94232767 100644
--- a/xbmc/cores/playercorefactory/PlayerCoreConfig.h
+++ b/xbmc/cores/playercorefactory/PlayerCoreConfig.h
@@ -20,6 +20,8 @@
 #include "system.h"
 #include "utils/log.h"
 
+#include "cores/GstPlayer/GstPlayer.h"
+
 class CPlayerCoreConfig
 {
 public:
@@ -87,6 +89,10 @@ public:
     {
       pPlayer = new KODI::RETRO::CRetroPlayer(callback);
     }
+    else if (m_type.compare("gstreamer") == 0)
+    {
+      pPlayer = new CGstPlayer(callback);
+    }
     else if (m_type.compare("external") == 0)
     {
       pPlayer = new CExternalPlayer(callback);
diff --git a/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp b/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp
index 477824e6957..533ed8e02b9 100644
--- a/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp
+++ b/xbmc/cores/playercorefactory/PlayerCoreFactory.cpp
@@ -25,6 +25,8 @@
 #include "utils/XMLUtils.h"
 #include <sstream>
 
+#include "cores/GstPlayer/GstPlayer.h"
+
 #define PLAYERCOREFACTORY_XML "playercorefactory.xml"
 
 CPlayerCoreFactory::CPlayerCoreFactory(const CProfileManager &profileManager) :
@@ -316,6 +318,11 @@ bool CPlayerCoreFactory::LoadConfiguration(const std::string &file, bool clear)
     m_vecCoreSelectionRules.clear();
 
     // Builtin players
+    CPlayerCoreConfig* GstPlayer = new CPlayerCoreConfig("GstPlayer", "gstreamer", nullptr);
+    GstPlayer->m_bPlaysAudio = true;
+    GstPlayer->m_bPlaysVideo = true;
+    m_vecPlayerConfigs.push_back(GstPlayer);
+
     CPlayerCoreConfig* VideoPlayer = new CPlayerCoreConfig("VideoPlayer", "video", nullptr);
     VideoPlayer->m_bPlaysAudio = true;
     VideoPlayer->m_bPlaysVideo = true;
@@ -351,6 +361,8 @@ bool CPlayerCoreFactory::LoadConfiguration(const std::string &file, bool clear)
         internaltype = "video";
       else if (type == "paplayer")
         internaltype = "music";
+      else if (type == "gstplayer")
+        internaltype = "gstreamer";
       else if (type == "externalplayer")
         internaltype = "external";
 
diff --git a/system/playercorefactory.xml b/system/playercorefactory.xml
index be6b721..c03699e 100644
--- a/system/playercorefactory.xml
+++ b/system/playercorefactory.xml
@@ -37,4 +37,8 @@
     <!-- pvr radio channels should be played by VideoPlayer because they need buffering -->
     <rule name="radio" filetypes="pvr" filename=".*/radio/.*" player="VideoPlayer" />
   </rules>
+
+  <rules action="prepend">
+    <rule filetypes="dts|mp3|wav|wave|oga|ogg|flac|m4a|mp2|m2a|ac3|mka|aac|ape|alac|mpg|vob|m4v|mkv|avi|divx|dat|flv|mp4|mov|wmv|asf|3gp|3g2|mpeg|mpe|rm|rmvb|ogm|ogv|stream|amr|au|mid|wv|pva|wtv|ts|m2ts" player="HiPlayer" />
+  </rules>
 </playercorefactory>
-- 
2.17.1
