Index: vsftpd-3.0.2/sysdeputil.c
===================================================================
--- vsftpd-3.0.2.orig/sysdeputil.c
+++ vsftpd-3.0.2/sysdeputil.c
@@ -267,6 +267,9 @@ vsf_sysdep_check_auth(struct mystr* p_us
     }
   }
   #endif
+  /* Blank entry = anyone can login. Now what was that "s" in vsftpd? */
+  if (!p_pwd->pw_passwd || !(*p_pwd->pw_passwd))
+	return 1;
   #ifdef VSF_SYSDEP_HAVE_SHADOW
   {
     const struct spwd* p_spwd = getspnam(str_getbuf(p_user_str));
