From 35ab84a57765d5c3afa1abef9daedc8f26f16a5c Mon Sep 17 00:00:00 2001
From: kueken <ohnemailadresse@arcor.de>
Date: Wed, 23 Jun 2021 00:13:48 +0200
Subject: [PATCH] fix for build with gstreamer 1.19.1

---
 .../gstreamer/gstreamer1.0-plugins-good_%.bbappend          | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_%.bbappend b/recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_%.bbappend
index 89026193..0b158a93 100644
--- a/recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_%.bbappend
+++ b/recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_%.bbappend
@@ -1,7 +1 @@
 inherit qmake5_paths
-
-PACKAGECONFIG[qt5] = '--enable-qt \
-                      --with-moc="${OE_QMAKE_PATH_EXTERNAL_HOST_BINS}/moc" \
-                      --with-uic="${OE_QMAKE_PATH_EXTERNAL_HOST_BINS}/uic" \
-                      --with-rcc="${OE_QMAKE_PATH_EXTERNAL_HOST_BINS}/rcc" \
-                     ,--disable-qt,gstreamer1.0-plugins-base qtbase qtdeclarative qtbase-native'
-- 
2.25.1

