From ebb40e69fedaa9c72a8168c5999059855e38b623 Mon Sep 17 00:00:00 2001
From: clnhub <42144823+clnhub@users.noreply.github.com>
Date: Mon, 27 Sep 2021 16:21:49 +0200
Subject: [PATCH] Add kernel 5.15(+) support

---
 core/rtw_br_ext.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/core/rtw_br_ext.c b/core/rtw_br_ext.c
index 43ffc31..c6172a8 100644
--- a/core/rtw_br_ext.c
+++ b/core/rtw_br_ext.c
@@ -17,8 +17,11 @@
 #ifdef __KERNEL__
 	#include <linux/if_arp.h>
 	#include <net/ip.h>
+	#include <linux/version.h>
+	#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
 	#include <net/ipx.h>
-	#ifndef _CHECKSM_IPV6_H
+	#endif
+	#ifndef _CHECKSM_IPV6_H
 	#include <net/ip6_checksum.h>
 	#endif
 	#include <linux/atalk.h>
@@ -68,6 +71,31 @@
 #define MAGIC_CODE_LEN	2
 #define WAIT_TIME_PPPOE	5	/* waiting time for pppoe server in sec */
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0))
+#define IPX_NODE_LEN	6
+struct ipx_address {
+	__be32  net;
+	__u8    node[IPX_NODE_LEN];
+	__be16  sock;
+};
+
+struct ipxhdr {
+	__be16			ipx_checksum __packed;
+#define IPX_NO_CHECKSUM	cpu_to_be16(0xFFFF)
+	__be16			ipx_pktsize __packed;
+	__u8			ipx_tctrl;
+	__u8			ipx_type;
+#define IPX_TYPE_UNKNOWN	0x00
+#define IPX_TYPE_RIP		0x01	/* may also be 0 */
+#define IPX_TYPE_SAP		0x04	/* may also be 0 */
+#define IPX_TYPE_SPX		0x05	/* SPX protocol */
+#define IPX_TYPE_NCP		0x11	/* $lots for docs on this (SPIT) */
+#define IPX_TYPE_PPROP		0x14	/* complicated flood fill brdcast */
+	struct ipx_address	ipx_dest __packed;
+	struct ipx_address	ipx_source __packed;
+};
+#endif
+
 /*-----------------------------------------------------------------
   How database records network address:
            0    1    2    3    4    5    6    7    8    9   10

