diff --git a/lib/python/Plugins/Extensions/HbbTV/plugin.py b/lib/python/Plugins/Extensions/HbbTV/plugin.py
--- a/lib/python/Plugins/Extensions/HbbTV/plugin.py
+++ b/lib/python/Plugins/Extensions/HbbTV/plugin.py
@@ -1057,6 +1057,8 @@
 			self._profile = self._applicationList[0]["profile"]
 			url = self._applicationList[0]["url"]
 		if url is None:
+			service = self._session.nav.getCurrentService()
+			info = service and service.info()
 			url = info.getInfoString(iServiceInformation.sHBBTVUrl)
 		return url
