diff --git a/lib/python/Plugins/Extensions/HbbTV/aitreader.py b/lib/python/Plugins/Extensions/HbbTV/aitreader.py
index f2c2195..4d0c63d 100755
--- a/lib/python/Plugins/Extensions/HbbTV/aitreader.py
+++ b/lib/python/Plugins/Extensions/HbbTV/aitreader.py
@@ -1,6 +1,13 @@
-import os, xml.dom.minidom
+import os, xml.dom.minidom, re
 from enigma import iServiceInformation
 
+RE_XML_ILLEGAL =	u'([\u0000-\u0008\u000b-\u000c\u000e-\u001f\ufffe-\uffff])' + \
+					u'|' + \
+					u'([%s-%s][^%s-%s])|([^%s-%s][%s-%s])|([%s-%s]$)|(^[%s-%s])' % \
+					(unichr(0xd800),unichr(0xdbff),unichr(0xdc00),unichr(0xdfff),
+					unichr(0xd800),unichr(0xdbff),unichr(0xdc00),unichr(0xdfff),
+					unichr(0xd800),unichr(0xdbff),unichr(0xdc00),unichr(0xdfff))
+
 DUMPBIN = "/usr/lib/enigma2/python/Plugins/Extensions/HbbTV/dumpait"
 class eAITSectionReader:
 	def __init__(self, demux, pmtid, sid):
@@ -71,7 +78,10 @@ class eAITSectionReader:
 			return False
 		if len(document) == 0:
 			return False
+		document = re.sub(RE_XML_ILLEGAL, "?", document)
+		document = re.sub("&", "+", document)
 		document = document.decode("cp1252").encode("utf-8")
+		document = "<URL>" + document + "</URL>"
 		#print document
 		self.mDocument = xml.dom.minidom.parseString(document)
 		return True
@@ -94,4 +104,3 @@ def unit_test(demux, pmtid, sid):
 	else:	print "no data!!"
 
 #unit_test('0', 0x17d4, 0x2b66)
-
