diff --git a/lib/python/Plugins/Extensions/HbbTV/plugin.py b/lib/python/Plugins/Extensions/HbbTV/plugin.py
index 50fa8fc..de520df 100644
--- a/lib/python/Plugins/Extensions/HbbTV/plugin.py
+++ b/lib/python/Plugins/Extensions/HbbTV/plugin.py
@@ -775,6 +775,12 @@ class HandlerHbbTV(Handler):
 		vodUri = data[hl:hl+uriLength]
 		self._handle_dump(self._cb_handleVODPlayerURI, opcode, vodUri)
 		self._vod_uri = vodUri
+		try:
+			if vodUri.find('/zdf/') >= 0:
+				tmpUri = MMSStreamURL().getLocationData(vodUri).strip()
+				if not strIsEmpty(tmpUri):
+					self._vod_uri = tmpUri
+		except: pass
 		return (0, "OK")
 
 	def doStop(self, restoreBeforeService=True, needStop=True):
